test/spam_test.go

```diff
@@ -17,7 +17,7 @@ func TestSpamAdversary(t *testing.T) {
 	t.Parallel()
 	const (
 		instanceCount = 2000
-		maxRounds     = 30
+		maxRounds     = 40
```

Comment:
I think this is from your debugging the problems with the commit case. Since you're not touching that here, probably revert this.